Source/Evolve/WarpXEvolve.cpp

```diff
@@ -494,7 +504,48 @@ WarpX::OneStep_nosub (Real cur_time)
     ExecutePythonCallback("particlescraper");
     ExecutePythonCallback("beforedeposition");
 
-    PushParticlesandDeposit(cur_time);
+    // with collisions placed in the middle of the position push and after the momentum push
+    if (m_collisions_split_position_push) {
+        // push particles (half position and full momentum)
+        PushParticlesandDeposit(
+            a_cur_time,
+            /*skip_current=*/true,
+            PositionPushType::FirstHalf,
+            MomentumPushType::Full
+        );
+
+        // communicate particle data
+        // FIXME Copy local communication from WarpX::HandleParticlesAtBoundaries
+        mypc->Redistribute();
+
+        // perform particle collisions
+        ExecutePythonCallback("beforecollisions");
+        mypc->doCollisions(a_step, a_cur_time, a_dt);
+        ExecutePythonCallback("aftercollisions");
+
+        // push particles (half position)
+        PushParticlesandDeposit(
+            a_cur_time,
+            /*skip_current=*/false,
```

Comment:
Reflect new name after #6303:
```suggestion
            /*skip_deposition=*/false,
```

Reply:
Resolved in [df7ffd7](https://github.com/BLAST-WarpX/warpx/pull/6275/commits/df7ffd784f989117779769f265dba79df38690d4).